OCPBUGS-1130: increase etcdGRPCRequestsSlow thresholds

tjungblu opened the pull request:
This replaces the upstream alert with something that's a lot less sensitive to bad etcd (fsync) latency.

tjungblu commented:
TODO update the runbook: https://github.com/openshift/runbooks/blob/master/alerts/cluster-etcd-operator/etcdGRPCRequestsSlow.md

Elbehery commented:
/lgtm 🎉

tjungblu commented:
@dusk125 / @hasbro17 PTAL

hasbro17 commented:
/lgtm
/retest-required

hasbro17 commented:
/jira refresh

wking commented on manifests/0000_90_etcd-operator_03_prometheusrule.yaml:
```diff
@@ -130,6 +119,17 @@ spec:
     rules:
+    - alert: etcdGRPCRequestsSlow
+      annotations:
+        description: 'etcd cluster "{{ $labels.%s }}": 99th percentile of gRPC requests is {{ $value }}s on etcd instance {{ $labels.instance }} for {{ $labels.grpc_method }} method.'
```
Are you sure you want `$labels.%s` here instead of `$labels.job`?

/hold

tjungblu replied:
good catch, thanks - fixing

wking commented on manifests/0000_90_etcd-operator_03_prometheusrule.yaml:
```diff
@@ -130,6 +119,17 @@ spec:
+        > 1
+      for: 30m
+      labels:
+        severity: critical
```
`critical` is [midnight-page territory][1].  And currently the only direct mitigation mentioned in the runbook is [running a defrag][2].  Perhaps the alert should pivot from "right now things are awful!" to "the past day has been pretty weak", and it could be a `warning` so folks could start thinking about provisioning faster disks in the near future?  Because it's hard for me to imaging rolling out of bed to run a defrag, or to scale up my control plane disks, and feeling like that was a healthy UX.

[1]: https://github.com/openshift/enhancements/blob/15aebc7549095cfc7eebc26d6460defc8e3927f6/enhancements/monitoring/alerting-consistency.md#critical-alerts
[2]: https://github.com/openshift/runbooks/blob/8513f248783bceddc3280e9387f7ac68e0ed0985/alerts/cluster-etcd-operator/etcdGRPCRequestsSlow.md#mitigation

tjungblu replied:
as mentioned above, I'm going to also update the runbook. Will send you the link to the updated one later today.
I would definitely want to get woken up if etcd takes longer than 1s to respond to anything. I'll revise the mitigation. 

> so folks could start thinking about provisioning faster disks

On BM that makes sense, in clouds you can easily scale up your VM type to unlock more IOPS or a faster CPU in a couple of minutes.

tjungblu replied:
https://github.com/openshift/runbooks/pull/69

wking replied:
> I would definitely want to get woken up if etcd takes longer than 1s to respond to anything.

Would you like to have had a `warning` ping the day before that response times were frequently up over 0.25s?  Or would that be too noisy, and you'd rather not hear about that and take the midnight page if/when it gets up to 1s?

tjungblu replied:
I would, if we assume that this is something that degrades slowly over time. The failures we observe in CI are transient, they come briefly and go away again. Would you pro-actively change anything in the build cluster when you see such warning the day before? 

Maybe we can find some folks over at SD to see whether their ROSA/ARO clusters would benefit from such warning. Not sure they would find that actionable as such and it would just drown in the rest of their 400 alerts a day.

wking commented on manifests/0000_90_etcd-operator_03_prometheusrule.yaml:
```diff
@@ -130,6 +119,17 @@ spec:
+        summary: etcd grpc requests are slow
+      expr: |
+        histogram_quantile(0.99, sum(rate(grpc_server_handling_seconds_bucket{job="etcd", grpc_method!="Defragment", grpc_type="unary"}[10m])) without(grpc_type))
+        > 1
```
We collect [a bunch of etcd performance metrics][1] in Telemetry.  Maybe we can process those, and possibly also correlate to other signs of cluster distress, to motivate particular thresholds?  No worries if we want to take a guess now, and circle back later when we have more time for analysis.

[1]: https://github.com/openshift/cluster-monitoring-operator/blame/492c01da6f0d16bdb046d7736da484e256c582df/Documentation/data-collection.md#L608-L629

tjungblu replied:
we have this in our backlog already in:
https://issues.redhat.com/browse/ETCD-144

feel free to add anything that's missing from your PoV.

tjungblu replied:
as for that threshold, we don't gather the grpc latencies sadly, so there's nothing much we can gather from telemeter unfortunately.

tjungblu commented:
@Elbehery can you please take a look again? I fixed a label for @wking and updated the runbook here:
https://github.com/openshift/runbooks/pull/69

tjungblu commented:
/hold cancel

tjungblu commented:
/cherry-pick release-4.11

Elbehery commented:
/lgtm

soltysh commented:
/test e2e-aws-ovn-serial

kikisdeliveryservice commented:
The current pass rate via sippy is 10 on ci and 20% on nightly for aws-ovn-serial jobs. I'm really not sure this job should be holding up this PR at this point.

tjungblu commented:
@hasbro17 and @dusk125 can help you out during the US working hours if this continues to fail.

kikisdeliveryservice commented:
`aws-ovn-serial ` heard me complain and proved me wrong :laughing:
